HeadersImports.md

```diff
@@ -0,0 +1,68 @@
+# Headers and Imports
+
+## Introduction
+
+Follow this set of guidelines when creating header files and importing them. The
+guidelines are designed to support a wide range of build systems and usage scenarios.
+
+In this document, the term `library` refers to a buildable package. In CocoaPods, it's a CocoaPod.
+In Swift Package Manager, it's a library target.
+
+## Header File Types and Locations
+
+* *Public Headers* - Headers that define the library's API. They should be located in
+  `FirebaseFoo/Sources/Public`. Any additions require a minor version update. Any changes or
+  deletions require a major version update.
+
+* *Public Umbrella Header* - A single header that includes the full library's public API located at
+  `FirebaseFoo/Sources/Public/FirebaseFoo.h`. This header should be included in
+   [Firebase.h](CoreOnly/Sources/Firebase.h).
+
+* *Library Internal Headers* - Headers that are only used by the enclosing library. These headers
+  should be located among the source files. [Xcode](https://stackoverflow.com/a/8016333) refers to
+  these as "Project Headers".
+
+* *Private Headers* - Headers that are available to other libraries in the repo, but are not part
+  of the public API. These should be located in `FirebaseFoo/Sources/Private`.
+  [Xcode](https://stackoverflow.com/a/8016333) and CocoaPods refer to these as "Private Headers".
+  Note that we are deprecating the usage of CocoaPods `private_headers` and should instead
+  publish them with `preserve_paths` and access them with a repo-relative import.
+
+## Imports
+
+* *Headers within the Library* - Use a repo-relative path. The one exception is
```

Comment:
You spend a bunch of time above breaking down the taxonomy of headers but then don't actually use those terms here. It's confusing. You called this "Library Internal Headers" above; use that term here?

Also, pull out the bit about public headers into its own bullet to match the structure above and to make it easier to skim.

Reply:
I was originally going to organize like that and then realized that thinking about creating and managing headers is much different and more complicated than being a consumer of them. I tried to make it more clear the document is organized like that.